src/com/company/exceptions/NumberOfArgsException.java

```diff
@@ -0,0 +1,19 @@
+package com.company.exceptions;
+
+public class NumberOfArgsException extends Exception{
+
+    private final int detail;
+
+    public NumberOfArgsException(int detail, String message) {
+        super(message);
+        this.detail = detail;
+    }
```

Comment:
Передавать сюда `message` с текстом "Неверное количество аргументов: " лишнее.
Это сообщение по сути уже есть в названии исключения `NumberOfArgsException`.
А название поля `detail`, наоборот, ни о чем не говорит. Лучше переименовать более понятно.

И еще хорошо передать правильное количество аргументов, если получится.

Как-то так:
```suggestion
    public NumberOfArgsException(int argsCount, int argsCountExpected) {
        super(String.format("Неверное количество аргументов: передано %d, ожидается %d", argsCount, argsCountExpected));
        this.argsCount = argsCount;
        this.argsCountExpected = argsCountExpected;
    }
```